assets/js/persistence.js

```diff
@@ -230,7 +247,7 @@ function restoreChoices(mode) {
             }
         }
 
-        if(getURLParam('qG').length > 0) {
+        if(getURLParam('q' + hashURLMap[hash]).length > 0) {
```

Comment:
I'm tempted to suggest offloading the 'q' into the `hashURLMap` to avoid this every time. What do you think?